python_modules/dagster/dagster/core/execution/context/output.py

```diff
@@ -291,7 +293,10 @@ def has_partition_key(self) -> bool:
                 "For more details: https://github.com/dagster-io/dagster/issues/7900"
             )
 
-        return self.step_context.has_partition_key
+        if self._step_context:
```

Comment:
Just a thought: might it be better to do this branching inside the constructor? Then it would only need to happen once, instead of every time these methods are called, and it would only need to be coded in one place?

Reply:
good idea

Reply:
hm on second thought - early access of partition key methods leads to partition errors when there are not partitions on the given run.

Reply:
hrm I guess. it checks tags, not key, which seems weird.